### PR TITLE
Mention existing integrations in ADR0007

### PR DESCRIPTION
**Pull request by MartinHjelmare**
- Add language about changing config YAML in a platform section for an existing integration:
https://github.com/home-assistant/architecture/pull/361#issuecomment-609826815

### CODE REVIEW

**Comment by thomasloven** [comment]
May discourage bugfixes, but I assume exceptions may still be granted on a case-by-case basis?

File: adr/0007-integration-config-yaml-structure.md [hunk 1]
```diff
@@ -33,6 +33,7 @@ C. Harder for Home Assistant to support confused users and contributors and hard
 ## Decision
 
 We limit the configuration YAML structure to one way for new integrations. We require all YAML configuration, if present, for an integration to be located under the integration domain key in configuration YAML, for all new integrations (point 2 above).
+For existing integrations we don't allow changes to YAML configuration in platform sections until the integration has been refactored and the configuration moved under the integration domain key.
```
Comment (balloob) [inline adr/0007-integration-config-yaml-structure.md:36]: If you want this on a new line, either add one before, or add 2 spaces to end of previous line. Now it's rendered as 1 paragraph.

**Comment by balloob** [comment]
Bug fixes mostly don't need YAML config changes.

**Comment by thomasloven** [comment]
The key word is "mostly". API changes may, for example.
But let's cross that bridge when we get there.

File: adr/0007-integration-config-yaml-structure.md (hunk 1 above)
Comment (MartinHjelmare) [reply adr/0007-integration-config-yaml-structure.md:36]: I didn't intend to add a paragraph. Should we add one?

**Comment by MartinHjelmare** [comment]
If the community wants to maintain an integration it will need to be updated to follow our current design sooner or later. Here we draw the line that changing configuration YAML requires an up to date integration structure.

**Pull request merged**